File: Array&String/Sorting/minimum_subsequence_in_non-increasing_order.py
class Solution:
    def minSubsequence(self, nums: list[int]) -> list[int]:
        if len(nums) == 1: return nums
        nums.sort(reverse = True)
        res, total, l = [nums[0]], sum(nums), nums[0]
        for i in range(1, len(nums)):
            if (l > total - l) : break
            l += nums[i]
            res.append(nums[i])
        return res

File: Array-String/Sorting/CPP/minimum_subsequence_in_non_increasing_order.cpp
#include <vector>
#include <numeric>
#include <functional>
using namespace std;

class Solution {
public:
    vector<int> minSubsequence(vector<int>& nums) {
        if (nums.size() == 1) return nums;
        sort(nums.begin(), nums.end(), greater<>());
        int total = accumulate(nums.begin(), nums.end(), 0);
        int l = nums[0];
        vector<int> res{nums[0]};
        for (int i = 1; i < nums.size(); i++) {
            if (l > total - l) break;
            l += nums[i];
            res.push_back(nums[i]);
        }
        return res;
    }
};